### PR TITLE
Add options to configure .github/workflows/meta.yml

### LINKED ISSUE #149: .meta.toml options to configure .github/workflows/meta.yml
Add options to:

- [ ] Disable GHA checks (i.e. dependencies -- [due to this issue](https://github.com/reinout/z3c.dependencychecker/issues/120))
- [ ] Add additional configurations to the file

### PR DESCRIPTION
**Pull request by ericof**
- [X] Disable GHA checks (i.e. dependencies -- [due to this issue](https://github.com/reinout/z3c.dependencychecker/issues/120))
- [X] Add additional configurations to the file

 Closes #149

### CODE REVIEW

File: config/README.md [hunk 1]
```diff
@@ -125,7 +125,32 @@ _your own configuration lines_
...
+[github]
+jobs = [
+    "qa",
+    "test",
+    "coverage",
+    "dependencies",
+    "release-ready",
+    "circular",
+    ]
+"""
```
Comment (gforcada) [inline config/README.md:141]:
```suggestion
```

Copy & pasted one line too much 😄

File: config/README.md [hunk 2]
```diff
@@ -125,7 +125,32 @@ _your own configuration lines_
...
+and set the enabled jobs with the `jobs` key.
+
+```toml
+[github]
+jobs = [
+    "qa",
+    "test",
+    "coverage",
+    "dependencies",
+    "release-ready",
```
Comment (gforcada) [inline config/README.md:138]:
```suggestion
    "release_ready",
```

Let's use the actual name, so we can simplify the template, there's already way too many (and more to come) options to remember how things should be named.

Maybe we can mention that the complete list is on https://github.com/plone/meta/tree/main/.github/workflows ? 🤔

File: config/README.md [hunk 3]
```diff
@@ -125,7 +125,32 @@ _your own configuration lines_
...
+Extend github workflow configuration with additional jobs
+by setting the values for the `additional_jobs` key.
+
+```toml
+[github]
+additional_jobs = """
+  another:
+    uses: org/repo/.github/workflows/file.yml@main
+"""
+```
```
Comment (gforcada) [inline config/README.md:153]: Would you mind adding another documentation section here for the `branch` variable?

I would like to eventually start using tags for this `branch` variable, so we can _safely_ evolve the workflows while at the same time ensuring we don't break existing workflows on repositories 🍀 

Maybe we can rename it `ref` ? 🤔

File: config/config-package.py [hunk 4]
```diff
@@ -32,6 +32,16 @@
 
 DOCKER_IMAGE = 'python:3.11-bullseye'
 
+GHA_DEFAULT_BRANCH = "main"
```
Comment (gforcada) [inline config/config-package.py:35]: If we go by the `ref` suggestion, then:

```suggestion
GHA_DEFAULT_REF = "main"
```

File: config/config-package.py [hunk 5]
```diff
@@ -32,6 +32,16 @@
...
+GHA_DEFAULT_JOBS = [
+    "qa",
+    "test",
+    "coverage",
+    "dependencies",
+    "release-ready",
```
Comment (gforcada) [inline config/config-package.py:41]: If you are ok with renaming it to match the actual name:

```suggestion
    "release_ready",
```

File: config/config-package.py [hunk 6]
```diff
@@ -335,6 +345,24 @@ def gha_workflows(self):
         folder = self.path / '.github' / 'workflows'
         folder.mkdir(parents=True, exist_ok=True)
         destination = folder / 'meta.yml'
+        options = self._get_options_for(
+            'github',
+            (
+                'branch',
```
Comment (gforcada) [inline config/config-package.py:351]: The `ref` suggestion:

```suggestion
                'ref',
```

File: config/config-package.py [hunk 7]
```diff
@@ -335,6 +345,24 @@ def gha_workflows(self):
...
+                'jobs',
+                'additional_jobs',
+            )
+        )
+        options.update(self._test_cfg())
```
Comment (gforcada) [inline config/config-package.py:356]: A copy & paste leftover? 🤔 I don't think we need the `_test_cfg` options her 😄 

```suggestion
```

File: config/config-package.py [hunk 8]
```diff
@@ -335,6 +345,24 @@ def gha_workflows(self):
...
+        if not options.get("branch"):
+            options['branch'] = GHA_DEFAULT_BRANCH
```
Comment (gforcada) [inline config/config-package.py:358]: The `ref` suggestion:

```suggestion
        if not options.get("ref"):
            options['branch'] = GHA_DEFAULT_REF
```

File: config/default/meta.yml.j2 [hunk 9]
```diff
@@ -11,15 +11,16 @@ on:
...
-    uses: plone/meta/.github/workflows/coverage.yml@main
-  dependencies:
-    uses: plone/meta/.github/workflows/dependencies.yml@main
-  release-ready:
-    uses: plone/meta/.github/workflows/release_ready.yml@main
-  circular:
-    uses: plone/meta/.github/workflows/circular.yml@main
+{% for job_name in jobs %}
+  %(job_name)s:
+    uses: plone/meta/.github/workflows/%( job_name | replace("-", "_") )s.yml@%( branch )s
```
Comment (gforcada) [inline config/default/meta.yml.j2:16]: Simplify the interpolation and `ref` suggestion:

```suggestion
    uses: plone/meta/.github/workflows/%(job_name)s.yml@%(ref)s
```

Oh 💡 now I see why you were doing the interpolation, for the line above... I would say we can live with an underscore there, maybe not so pretty, but it's CI, so who cares 😅 we care much more about the green tick that should appear next to it 😆

**Review by gforcada — COMMENTED** [review]
Great job, thanks! 🌟 

I have 2 minor wishes (`branch` -> `ref` and the simplification of the dash to underscore).

Awesome!! 💯

File: config/README.md [hunk 10]
```diff
@@ -125,7 +125,32 @@ _your own configuration lines_
...
+    ]
+"""
+```
+
+Extend github workflow configuration with additional jobs
+by setting the values for the `additional_jobs` key.
+
+```toml
+[github]
+additional_jobs = """
```
Comment (gforcada) [inline config/README.md:149]: Could we rename that, for consistency across the other files, `extra_lines` ? 🍀 🤞🏾

File: config/config-package.py [hunk 11]
```diff
@@ -335,6 +345,24 @@ def gha_workflows(self):
         folder = self.path / '.github' / 'workflows'
         folder.mkdir(parents=True, exist_ok=True)
         destination = folder / 'meta.yml'
+        options = self._get_options_for(
+            'github',
+            (
+                'branch',
+                'jobs',
+                'additional_jobs',
```
Comment (gforcada) [inline config/config-package.py:353]: As suggested, we could rename this `extra_lines` 🍀 

```suggestion
                'extra_lines',
```

File: config/README.md (hunk 3 above)
Comment (ericof) [reply config/README.md:163]: Done

**Comment by ericof** [comment]
@gforcada Changes applied

File: config/README.md [hunk 12]
```diff
@@ -125,7 +125,42 @@ _your own configuration lines_
...
+```toml
+[github]
+ref = "master"
+```
+
+In the previous example, all GitHub workflows would come from
+the `master` branch, instead of the default `main` branch.
+
+Extend github workflow configuration with additional jobs
+by setting the values for the `additional_jobs` key.
```
Comment (gforcada) [inline config/README.md:155]:
```suggestion
by setting the values for the `extra_lines` key.
```

File: config/default/meta.yml.j2 [hunk 13]
```diff
@@ -11,15 +11,16 @@ on:
...
-    uses: plone/meta/.github/workflows/dependencies.yml@main
-  release-ready:
-    uses: plone/meta/.github/workflows/release_ready.yml@main
-  circular:
-    uses: plone/meta/.github/workflows/circular.yml@main
+{% for job_name in jobs %}
+  %(job_name)s:
+    uses: plone/meta/.github/workflows/%(job_name)s.yml@%(ref)s
+{% endfor %}
+%(additional_jobs)s
```
Comment (gforcada) [inline config/default/meta.yml.j2:18]:
```suggestion
%(extra_lines)s
```

**Review by gforcada — COMMENTED** [review]
I forgot two places that needed replacement :)

**Review by gforcada — APPROVED** [review]

**Comment by gforcada** [comment]
Great job!! 😄 

Now I need to see how to mix this with the #65 🍀

**Pull request merged**